File: 2520. Count the Digits That Divide a Number.java
class Solution {
    public int countDigits(int num) {        
        int count = 0, t = num;

        while(t > 0){
            int temp = t % 10;
            t /= 10;
            if(num % temp == 0) count++;
        }

        return count;
    }
}

File: 2520. Count the Digits That Divide a Number.ts
function countDigits(num: number): number {
    let result = 0, a = num;

    while(a > 0){
        let temp = a % 10;
        a = (a - temp)/10;
        if(num % temp == 0){
            result++;
        }
    }

    return result;
};
